### ISSUE TITLE
Reconfiguring table shown as unavailable after restart

### ISSUE THREAD

**Issue by danielmewes** [rethinkdb/rethinkdb#4383]
On a two server cluster, I observed the following behavior on `raft`:
1. Create a table and insert some data (e.g. 1 M rows, must be enough for backfilling to take a while)
2. Configure the table to two replicas
3. The table will remain ready for writes while the backfill is going on.
4. Now shut down and restart both servers.
5. The table will now show in `table_status` with `"ready_for_writes": false`. However writing to the table actually appears to be working, so this just seems to be a case of `table_status` being wrong.

I have the _debug_table_status output for this table, but it had 16 shards and the output is pretty long. I'll try to reproduce this with a simpler configuration later.

This is not super critical because only the status appears to be inaccurate.

**Comment by danielmewes** [comment]
Actually my description was wrong. It seems that the table is actually unavailable for writes either for a subset of shards or intermittently (not sure which):
```
Cannot perform write: primary replica error: The primary replica isn't connected to a quorum of replicas. The write was not performed in:
```

**Comment by danielmewes** [comment]
Going to move this into polish. I think this is acceptable unless there's an easy fix for it.

**Comment by timmaxw** [comment]
Have you confirmed that the table becomes available for writes (both as reported in `table_status`, and in reality) within a reasonable amount of time?

**Comment by danielmewes** [comment]
Yes, when all backfills are done the table becomes available.

**Comment by danielmewes** [comment]
This might only happen if the table is also sharded. Try configuring it to `{shards: 16, replicas: 1}` first, waiting until it's done, and then reconfiguring to `{shards: 16, replicas: 2}` before restarting the servers.

**Comment by timmaxw** [comment]
In CR 3035.

**Comment by timmaxw** [comment]
Fixed in 3dac99483c181c0e69e67e8c99d002cbe5d98b7b.
